server/remote_cache/digest/digest_amd64_test.go

```diff
@@ -0,0 +1,71 @@
+//go:build amd64
+
+package digest_test
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"hash"
+	"io"
+	"math/rand"
+	"testing"
+
+	sha256simd "github.com/minio/sha256-simd"
+	"github.com/stretchr/testify/require"
+)
+
+func hasherWithServer() hash.Hash {
+	server := sha256simd.NewAvx512Server()
+	return sha256simd.NewAvx512(server)
+}
```

Comment:
in real usage, would we reuse this server across requests? wonder if the `server` should be declared as a top-level var

Reply:
https://github.com/minio/sha256-simd/blob/master/README.md#support-for-avx512

```
Due to this different way of scheduling, we decided to use an explicit method to instantiate the AVX512 version. Essentially one or more AVX512 processing servers ([Avx512Server](https://github.com/minio/sha256-simd/blob/master/sha256blockAvx512_amd64.go#L294)) have to be created whereby each server can hash over 3 GB/s on a single core. An hash.Hash object ([Avx512Digest](https://github.com/minio/sha256-simd/blob/master/sha256blockAvx512_amd64.go#L45)) is then instantiated using one of these servers and used in the regular fashion:
```

I think the expectation here is to create 1 server for each core? there are not a lot of examples 🤔